src/libraries/System.Private.CoreLib/src/System/Globalization/CharUnicodeInfo.cs

```diff
@@ -250,6 +250,77 @@ private static double GetNumericValueNoBoundsCheck(uint codePoint)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static char ToUpper(char codePoint)
+        {
+            nuint offset = GetCategoryCasingTableOffsetNoBoundsChecks((uint)codePoint);
+
+            // The offset is specified in shorts:
+            // Get the 'ref short' corresponding to where the addend is, read it as a signed 16-bit value, then add
+
+            ref short rsStart = ref Unsafe.As<byte, short>(ref MemoryMarshal.GetReference(UppercaseValues));
+            ref short rsDelta = ref Unsafe.Add(ref rsStart, (nint)offset);
+            nint delta = (BitConverter.IsLittleEndian) ? rsDelta : BinaryPrimitives.ReverseEndianness(rsDelta);
+            return (char)(delta + (nint)codePoint);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static uint ToUpper(uint codePoint)
+        {
+            if (UnicodeUtility.IsBmpCodePoint(codePoint))
```

Comment:
Removing this _if_ block will improve method performance and codegen. The rest of the method handles BMP code points just fine.

(But remember to keep the _IsValidCodePoint_ check below!)

Reply:
Yes, this was mistakenly left over. I already removed it from another place. Thanks!